enterprise/tailnet/pgcoord_test.go

```diff
@@ -625,17 +601,36 @@ func newTestClient(t *testing.T, coord agpl.Coordinator, agentID uuid.UUID, id .
 	return c
 }
 
-func assertHasDERPs(t *testing.T, nodes []*agpl.Node, expected ...int) {
-	if !assert.Len(t, nodes, len(expected), "expected %d node(s), got %d", len(expected), len(nodes)) {
+func assertEventuallyHasDERPs(ctx context.Context, t *testing.T, c *testConn, expected ...int) {
+	t.Helper()
+	for {
+		nodes := c.recvNodes(ctx, t)
+		if len(nodes) != len(expected) {
+			t.Logf("expected %d, got %d nodes", len(expected), len(nodes))
+			continue
+		}
+
+		derps := make([]int, 0, len(nodes))
+		for _, n := range nodes {
+			derps = append(derps, n.PreferredDERP)
+		}
+		for _, e := range expected {
+			if !contains(derps, e) {
+				t.Logf("expected DERP %d to be in %v", e, derps)
+				continue
+			}
+		}
 		return
 	}
-	derps := make([]int, 0, len(nodes))
-	for _, n := range nodes {
-		derps = append(derps, n.PreferredDERP)
-	}
-	for _, e := range expected {
-		assert.Contains(t, derps, e, "expected DERP %v, got %v", e, derps)
+}
+
+func contains(s []int, i int) bool {
```

Comment:
nit: maybe [slices.Contains](https://pkg.go.dev/golang.org/x/exp/slices#Contains)?

Reply:
We should remove our `slices` package maybe if this has all the ones we made.